### PR TITLE
feat(pipelines): remove context/outputs/tasks on /pipelines?expand=fa…

### PR DESCRIPTION
**Pull request by anotherchrisberry**
…lse calls

Should not be merged/deployed without https://github.com/spinnaker/deck/pull/5141

Trims the execution payload by a lot in most cases, deferring to Deck to fetch the details as needed.

There is probably more that can be trimmed. This seems to cut ~65% of the payload for some applications, but I suspect there's more that can go.

Update: this is cutting ~93% of the payload on our heavier applications.

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 1]
```diff
@@ -337,11 +339,13 @@ class TaskController {
 
     if (!expand) {
       allPipelines.each { pipeline ->
+        if (pipeline.trigger instanceof PipelineTrigger) {
+          ((PipelineTrigger) pipeline.trigger).parentExecution.getStages().clear()
```
Comment (anotherchrisberry) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: So, interesting story, this line definitely gets called but also definitely seems to not make any difference in the response, as I will still see all the stages in the `trigger.parentExecution`... still trying to figure out why that is the case...

**Comment by robzienert** [comment]
I'm curious about the rollout here. The description makes it sound like deck & orca will be in lockstep with each other on this change. Are we concerned with backwards compatibility? What will the behavior be while red/blacking?

**Comment by ajordens** [comment]
Agree with Rob. If the param is ignored, is it problematic to deck? If not that we don’t really need to be careful about order of rollout.

**Comment by anotherchrisberry** [comment]
We just need Deck to go out before Orca here. Deck is fine to go without this.

**Pull request merged**